### PR TITLE
Add core & sql managed rules to WAF

### PR DESCRIPTION
**Pull request by kenoir**
## Who is this for?

This change is intended to block a wider array of malicious requests to wellcomecollection.org in order to reduce unnecessary alarms and increase protection to the underlaying services.

## What is it doing for them?

We add `AWSManagedRulesSQLiRuleSet`, `AWSManagedRulesCommonRuleSet` and `AWSManagedRulesKnownBadInputsRuleSet`, to block requests that match recognised OWASP vulnerabilities and include potential (useless) SQL injection attacks that we've seen in alerts.

## `terraform plan`

Targeting the prod stack:

`terraform plan -target module.prod_wc_org_cloudfront_distribution`

Output:
```
  # module.prod_wc_org_cloudfront_distribution.aws_wafv2_web_acl.wc_org will be updated in-place
  ~ resource "aws_wafv2_web_acl" "wc_org" {
        id          = "316ddf50-ea9e-4555-826d-e0a4d1380ead"
        name        = "weco-cloudfront-acl-prod"
        tags        = {}
        # (6 unchanged attributes hidden)

      - rule {
          - name     = "managed-ip-blocking" -> null
          - priority = 1 -> null

          - override_action {
              - none {}
            }

          - statement {
              - managed_rule_group_statement {
                  - name        = "AWSManagedRulesAmazonIpReputationList" -> null
                  - vendor_name = "AWS" -> null
                }
            }

          - visibility_config {
              - cloudwatch_metrics_enabled = true -> null
              - metric_name                = "weco-cloudfront-acl-ip-block-prod" -> null
              - sampled_requests_enabled   = true -> null
            }
        }
      + rule {
          + name     = "core-rule-group"
          + priority = 4

          + override_action {
              + count {}
            }

          + statement {
              + managed_rule_group_statement {
                  + name        = "AWSManagedRulesCommonRuleSet"
                  + vendor_name = "AWS"
                }
            }

          + visibility_config {
              + cloudwatch_metrics_enabled = true
              + metric_name                = "weco-cloudfront-acl-core-prod"
              + sampled_requests_enabled   = true
            }
        }
      + rule {
          + name     = "known-bad-inputs-rule-group"
          + priority = 6

          + override_action {
              + count {}
            }

          + statement {
              + managed_rule_group_statement {
                  + name        = "AWSManagedRulesKnownBadInputsRuleSet"
                  + vendor_name = "AWS"
                }
            }

          + visibility_config {
              + cloudwatch_metrics_enabled = true
              + metric_name                = "weco-cloudfront-acl-known-bad-inputs-prod"
              + sampled_requests_enabled   = true
            }
        }
      + rule {
          + name     = "managed-ip-blocking"
          + priority = 1

          + override_action {
              + none {}
            }

          + statement {
              + managed_rule_group_statement {
                  + name        = "AWSManagedRulesAmazonIpReputationList"
                  + vendor_name = "AWS"
                }
            }

          + visibility_config {
              + cloudwatch_metrics_enabled = true
              + metric_name                = "weco-cloudfront-acl-ip-block-prod"
              + sampled_requests_enabled   = true
            }
        }
      + rule {
          + name     = "sqli-rule-group"
          + priority = 5

          + override_action {
              + count {}
            }

          + statement {
              + managed_rule_group_statement {
                  + name        = "AWSManagedRulesSQLiRuleSet"
                  + vendor_name = "AWS"
                }
            }

          + visibility_config {
              + cloudwatch_metrics_enabled = true
              + metric_name                = "weco-cloudfront-acl-sqli-prod"
              + sampled_requests_enabled   = true
            }
        }

        # (5 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

## How do we test this?

- [x] Deploy this change to the e2e stack, run the e2e tests to catch any obvious problems
- [ ] Deploy this change with the action set to count 
        In order that we can deploy to production and identify potential impact with real traffic before switching to block.

> [!NOTE]
> The `AWSManagedRulesCommonRuleSet` has the potential to match valid requests as it does things like match large request bodies and long urls. Unlikely to be an issue but it's something we can test for so we should probably do that.

### CODE REVIEW

**Comment by kenoir** [comment]
Part of https://github.com/wellcomecollection/wellcomecollection.org/issues/10511

File: cache/modules/wc_org_cloudfront/waf.tf [hunk 1]
```diff
@@ -150,6 +150,75 @@ resource "aws_wafv2_web_acl" "wc_org" {
     }
   }
 
+  // See: https://docs.aws.amazon.com/waf/latest/developerguide/aws-managed-rule-groups-baseline.html#aws-managed-rule-groups-baseline-crs
+  rule {
+    name     = "core-rule-group"
```
Comment (jamieparkinson) [inline cache/modules/wc_org_cloudfront/waf.tf:155]: I'm not attached to this suggestion at all but you could consider commonising the managed rules sort of like 
```hcl
locals {
  managed_rule_groups = {
    // link to docs
    core = {
      name = "AWSManagedRulesCommonRuleSet"
      priority = 4
    }
    // link to docs
    sqli = {
       name = "AWSManagedRulesSQLiRuleSet"
       priority = 5
    }
    // link to docs
    known-bad-inputs = {
      name = "AWSManagedRulesKnownBadInputsRuleSet"
      priority = 6
    }
  }
}

// ...

dynamic "rule" {
  for_each = local.managed_rule_groups
  content {
    name = "${rule.key}-rule-group"
    priority = rule.value.priority
 
    override_action {
      count {}
    }
    
    statement {
      managed_rule_group_statement {
        name = rule.value.name
        vendor_name = "AWS"
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = true
      sampled_requests_enabled = true
      metric_name = "weco-cloudfront-acl-${rule.key}-${var.namespace}"
    }
  }  
}
```

Comment (jamieparkinson) [reply cache/modules/wc_org_cloudfront/waf.tf:155]: As soon as I wrote it I changed my mind on this, it just adds complexity, ignore

**Review by jamieparkinson — APPROVED** [review]

File: cache/modules/wc_org_cloudfront/waf.tf (hunk 1 above)
Comment (kenoir) [reply cache/modules/wc_org_cloudfront/waf.tf:155]: A few more of these and this could be the way to go though.

**Pull request merged**